scripts/webpack/webpack-resources.js

```diff
@@ -1,11 +1,55 @@
 const webpack = require('webpack');
 const path = require('path');
+const fs = require('fs');
+const resolve = require('resolve');
 const merge = require('../tasks/merge');
 const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 
 const webpackVersion = require('webpack/package.json').version;
 console.log(`Webpack version: ${webpackVersion}`);
 
+let isValidEnv = false;
+
+function validateEnv() {
+  if (!isValidEnv) {
+    try {
+      const resolvedPolyfill = resolve.sync('react-app-polyfill/ie11', { basedir: process.cwd() });
+      isValidEnv = !!resolvedPolyfill;
+    } catch (e) {
+      console.error('Please make sure the package "react-app-polyfill" is in the package.json dependencies');
+      process.exit(1);
+    }
+  }
+}
+
+/**
+ * Prepends the entry points with a react 16 compatible polyfill but only for sites that have react as a dependency
+ */
+function prependEntryWithPolyfill(entry) {
+  const packageJson = JSON.parse(fs.readFileSync(path.join(process.cwd(), 'package.json'), 'utf-8'));
+  if (
+    (packageJson.dependencies && Object.keys(packageJson.dependencies).includes('react')) ||
```

Comment:
What about peer deps?

Reply:
I shouldnt dictate the polyfill for our library consumers. I think we should leave the peer deps alone..